Automattic/wp-calypso
# SSR: Use document-head to populate `title`, `meta`s, and `link`s; drop `Helmet`, take two

Take two for #7600, which had to be reverted (#8420) because of #8419.

The actual problem was that <code>client/boot</code> originally rendered a <code>Layout</code> component for non-isomorphic sections, and it would find out if a section was isomorphic or not by looking at the Redux subtree. For isomorphic sections, a Redux state tree was created on the server and passed on to the client, but only if there was also a <code>Layout</code> element tree rendered on the server; otherwise, it would just fall back to <code>false</code>. This means that even though <code>signup</code> was set to isomorphic, <code>client/boot</code> wasn't aware since it didn't get any Redux tree from the server as there was no <code>Layout</code> rendered on the server.

With #7600, a Redux tree would be passed to the client no matter what, with `isomorphic` obviously set to `true` for all `isomorphic` sections, including `signup`. Hence, `client/boot` didn't render a `Layout` element. This caused an error with <code>signup</code>'s <code>start</code> middleware <a href="https://github.com/Automattic/wp-calypso/blob/f5b4197a578297ad3a30b23e3f4be072579ffddb/client/signup/controller.js#L87">attempting</a> to unmount the element tree found in <code>secondary</code> (when there wasn't one).

To test:
* See #7600 
* Additionally, make sure that logged-out http://calypso.localhost:3000/start works.
